packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -1,19 +1,157 @@
 import React, { useState, useEffect, useCallback, useRef } from "react"
-
+import { formatDistance } from "date-fns"
+import { Link } from "gatsby-interface"
 import getBuildInfo from "../utils/getBuildInfo"
+import IndicatorButton from "./IndicatorButton"
+import {
+  gatsbyIcon,
+  logsIcon,
+  failedIcon,
+  infoIcon,
+  linkIcon,
+  successIcon,
+} from "./icons"
 import Style from "./Style"
 
-import GatsbyIndicatorButton from "./GatsbyIndicatorButton"
-import LinkIndicatorButton from "./LinkIndicatorButton"
-import InfoIndicatorButton from "./InfoIndicatorButton"
-
 const POLLING_INTERVAL = process.env.GATSBY_PREVIEW_POLL_INTERVAL || 3000
 
-export default function Indicator({ children }) {
+export function PreviewIndicator({ children }) {
+  return (
+    <>
+      <Style />
+      <div
+        data-testid="preview-status-indicator"
+        data-gatsby-preview-indicator="root"
+        aria-live="assertive"
+      >
+        {React.Children.map(children, (child, i) =>
+          React.cloneElement(child, { ...child.props, buttonIndex: i })
+        )}
+      </div>
+    </>
+  )
+}
+
+export function GatsbyIndicatorButton(props) {
+  return <IndicatorButton testId="gatsby" iconSvg={gatsbyIcon} {...props} />
+}
+
+const copySuccessTooltip = (
+  <>
+    {successIcon}
+    {`Link copied`}
+  </>
+)
+
+export function LinkIndicatorButton(props) {
+  const [linkButtonCopyProps, setLinkButtonCopyProps] = useState()
+
+  const copyLinkClick = () => {
+    setLinkButtonCopyProps({
+      tooltipContent: copySuccessTooltip,
+      overrideShowTooltip: true,
+    })
+
+    setTimeout(() => {
+      setLinkButtonCopyProps({
+        tooltipContent: copySuccessTooltip,
+        overrideShowTooltip: false,
+      })
+      // We want the tooltip to linger for two seconds to let the user know it has been copied
+    }, 2000)
+
+    setTimeout(() => {
+      setLinkButtonCopyProps({ tooltipContent: `Copy Link` })
+      // The tooltips fade out, in order to make sure that the text does not change
+      // while it is fading out we need to wait a bit longer than the time used above.
+    }, 2400)
+
+    navigator.clipboard.writeText(window.location.href)
+  }
+
+  return (
+    <IndicatorButton
+      testId={`link`}
+      iconSvg={linkIcon}
+      onClick={copyLinkClick}
+      {...linkButtonCopyProps}
+      {...props}
+    />
+  )
+}
+
+export function InfoIndicatorButton(props) {
+  return <IndicatorButton testId="info" iconSvg={infoIcon} {...props} />
+}
+
+export function BuildErrorIndicatorTooltip({ siteId, orgId, buildId }) {
+  return (
+    <>
+      {failedIcon}
+      {`Unabled to build preview`}
+      <Link
```

Comment:
Decided to use a link here for A11y. We should be able to use the onClick for click tracking!

cc @DanielSLew

Reply:
I'm not too familiar with this `Link` component, but does it open up a new window or navigate on the current window? The behaviour we'd want is for it to open a new window/tab.

Reply:
Link from gatsby-interface uses [Gatsby Link](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-link/) under the hood. For Gatsby sites you only want to use Link for local navigation within the Gatsby site, but gatsby-interface has logic for that built in https://github.com/greglobinski/gatsby-interface/blob/master/src/components/Link/Link.tsx

Reply:
Props are spread onto the underlying BaseLink so I believe you can add `<Link target="_blank">` if you want it to open in a new tab.

Reply:
Yup, what Tyler said! Additionally, that `target` attribute is defaulted to `"_blank"`

https://github.com/gatsby-inc/gatsby-interface/blob/main/src/components/BaseAnchor/BaseAnchor.tsx#L8

Reply:
`gatsby-interface` has peer deps I didn't notice at first so I've changed this to an `a`